tracer/src/Datadog.Trace/Logging/Internal/ExceptionRedactor.cs

```diff
@@ -0,0 +1,449 @@
+using System;
+using System.Collections;
+using System.Diagnostics;
+using System.Globalization;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Security;
+using System.Text;
+using Datadog.Trace.ClrProfiler;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.Logging.Internal;
+
+internal static class ExceptionRedactor
+{
+    internal const string StackFrameAt = "   at ";
+    internal const string Redacted = "REDACTED";
+    private const string InFileLineNum = "in {0}:line {1}";
+
+    /// <summary>
+    /// Redacts a stacktrace by replacing non-Datadog and non-BCL stack frames with REDACTED.
+    /// Uses code based on the <a href="https://github.com/dotnet/runtime/blob/v7.0.2/src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs" >.NET Core <c>StackTrace</c></a> code and the
+    /// <a href="https://referencesource.microsoft.com/#mscorlib/system/diagnostics/stacktrace.cs" >.NET Framework <c>StackTrace</c></a>
+    /// Records the _type_ of the Exception instead of the exception message
+    /// </summary>
+    /// <param name="exception">The exception to generate the redacted stack trace for</param>
+    /// <returns>The redacted stack trace</returns>
+    public static string Redact(Exception exception)
+    {
+        var exceptionType = exception.GetType().FullName ?? "Unknown Exception";
+        var stackTrace = new StackTrace(exception);
+
+        var stackFrameCount = stackTrace.FrameCount;
+        if (stackFrameCount == 0)
+        {
+            return exceptionType;
+        }
+
+        var sb = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
+        sb.AppendLine(exceptionType);
+
+        RedactStackTrace(sb, stackTrace);
+
+        return StringBuilderCache.GetStringAndRelease(sb);
+    }
+
+    // internal for testing
+    internal static void RedactStackTrace(StringBuilder sb, StackTrace stackTrace)
+    {
+        for (var i = 0; i < stackTrace.FrameCount; i++)
+        {
+            var frame = stackTrace.GetFrame(i);
+            var methodInfo = frame?.GetMethod();
+            if (frame is null || methodInfo is null)
+            {
+                continue;
+            }
+
+            if (ShouldRedactFrame(methodInfo))
+            {
+                sb.Append(StackFrameAt);
+                sb.AppendLine(Redacted);
+                continue;
+            }
+
+            AppendFrame(sb, frame);
+        }
+    }
+
+    private static bool ShouldRedactFrame(MethodBase mb)
+        => mb.DeclaringType switch
+        {
+            null => true, // "global" function
+            // NOTE: Keep this in sync with InstrumentationDefinitions SourceGenerator implementation in Sources.BuildInstrumentedAssemblies.IsKnownAssemblyPrefix()
+            { Assembly.FullName: { } name } => !(name.StartsWith("Datadog.", StringComparison.Ordinal)
+                                              || name.StartsWith("mscorlib,", StringComparison.Ordinal)
```

Comment:
copying over the same comment from the other PR
```suggestion
                                              || name.StartsWith("mscorlib,", StringComparison.Ordinal) // note that this uses ',' not '.' as it's the full assembly name
```